src/python/WMComponent/RetryManager/Modifier/MemoryModifier.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+
+"""
+_MemoryModifier_
+
+config.RetryManager.section_('MemoryModifier')
+config.RetryManager.MemoryModifier.section_('default')
+config.RetryManager.MemoryModifier.default.maxMemory = 2000 # Memory/cpu in MB
+config.RetryManager.MemoryModifier.section_('merge')
+config.RetryManager.MemoryModifier.merge.maxMemory = 3000 # Memory/cpu in MB
+"""
+
+import pickle
+from WMCore.WMSpec.WMWorkload import WMWorkloadHelper
+from WMComponent.RetryManager.Modifier.BaseModifier import BaseModifier
+import pickle
+
+
+class MemoryModifier(BaseModifier):
+
+    def changeSandbox(self, jobPKL, newMemory):
+        """
+        _changeSandbox_
+
+        Modifies the parameter maxPSS in the sandbox. This is a change that applies for all jobs in that workflow that remain to be submitted 
+        """
+
+        workload = self.getWorkload(jobPKL)
+        workHelper = WMWorkloadHelper(workload)
+
+        for task in workHelper.getAllTasks():
+            task.setMaxPSS(newMemory)
+
+        self.setWorkload(workload)
+
+        return
+
+    def changeJobPkl(self, jobPKL, newMemory):
+        """
+        Modifies the pklFile job.pkl by changing the estimatedMemoryUsage to a new_memory value
+
+        """
+        jobPKL['estimatedMemoryUsage'] = newMemory
+        pklFile = '{}/job.pkl'.format(jobPKL['cache_dir']) 
+        self.savePKL(pklFile,data)
+
+
+    def checkNewJobPkl(pklFile):
+        with open(pklFile, 'rb') as file:
+            data = pickle.load(file)
+        print (data['estimatedMemoryUsage'])
+
+    def getNewMemory(self, jobPKL):
+        maxMemPerCore = self.getModifierParam(job['jobType'], 'maxMemory')
+        #Finds job.pkl file
+
+        currentMem = jobPKL['estimatedMemoryUsage']
+        currentMemPerCore = currentMem/jobPKL['numberOfCores']
+        newMemPerCore = currentMemPerCore * 1.5
```

Comment:
El valor 1.5 puede entrar como parametro de la configuracion. Tipo:

config.RetryManager.MemoryModifier.Processing.addMemory = 2000
config.RetryManager.MemoryModifier.Processing.multiplyMemory = 0.5
Entonces puede ser algo asi

if self.addMemory
    newMemPerCore = currentMemPerCore + addMemory
elif self.multiplyMemory:
    newMemPerCore = currentMemPerCore * multiplyMemory
else:
    logging.info('New memory was not specified in the configuration')